server/aesdsocket.c

```diff
@@ -0,0 +1,394 @@
+        return -1;
+    }
+
+    // Flag to check if \n was received
+    uint8_t newline_flag = 0;
+    int byte_count = 0;
+    int datapacket = 0;
+    int final_size = 1;
+    int realloc_int = 0;
+    static int file_size = 0;
+
+    // While any signal has not been encountered yet
+    while(1){
+
+        new_fd = accept(sd , &client_addr, &addr_size);
+        // Error check 
+        if (new_fd == -1){
+            syslog(LOG_ERR , "accept\n");
+            // Try waiting for a connect() request from another remote
+            continue;
+        } 
+
+        // Print IP address
+        // The inet_toa function requires a sockaddr_in structure
+        // Reference: https://stackoverflow.com/questions/3060950/how-to-get-ip-address-from-sock-structure-in-c
+        struct sockaddr_in * ip_client = (struct sockaddr_in *)&client_addr;
+        syslog(LOG_INFO , "Accepted connection from %s\n", inet_ntoa(ip_client->sin_addr));  
+
+        while(!newline_flag){
+            byte_count = recv(new_fd , recv_buffer , BUFFER_SIZE , 0);
+            // printf("Byte count:%d\n", byte_count);
+            if ( byte_count == -1){
+                syslog(LOG_ERR , "recv\n");
+                return -1;
+            }
+            else{
+                final_size++;
+                for(datapacket = 0; datapacket < BUFFER_SIZE; datapacket++){
+                    if(recv_buffer[datapacket] == '\n'){
+                        newline_flag = 1;
+                        break;
+                    }
+                }
+
+                memcpy((write_buffer + (BUFFER_SIZE * realloc_int)) , recv_buffer , BUFFER_SIZE);
+                // printf("Buffer:%s\n", write_buffer);
+
+                if(!newline_flag){
+                    write_buffer = (char *)realloc(write_buffer , (sizeof(char) * (BUFFER_SIZE * final_size)));
+                    // If realloc fails
+                    if (write_buffer == NULL){
+                        syslog(LOG_ERR , "realloc\n");
+                    }
+                    else{
+                        realloc_int++;
+                    }                    
+                }
+            }
+        }
+
+        // Reset the flag
+        newline_flag = 0;
+
+        // Now the data packet is acquired
+
+        // Write the data packet to the file
+        int buffer_len = ((datapacket + 1) + (BUFFER_SIZE * realloc_int));
+        int write_len = write(fd , write_buffer , buffer_len);
+        // Error check
+        if(write_len == -1){
+            syslog(LOG_ERR , "write\n");
+            return -1;
+        }
+        // printf("Write len:%d\n", write_len);
+
+        file_size += write_len;
+
+        // printf("Size: %d\n", file_size);
+
+        // Reset the reallocation tracking flag and the final size tracking flag
+        realloc_int = 0, final_size = 1;
+
+        // Allocate memory for the send buffer
+        char *send_buffer = (char *)malloc(sizeof(char) * file_size);
```

Comment:
@KhyatiSatta what assumption are you making here about the total size of the heap relative to the size of all packets sent?